src/sentry/integrations/pagerduty/utils.py

```diff
@@ -51,9 +54,27 @@ def send_incident_alert_notification(
     new_status: IncidentStatus,
 ) -> None:
     integration_id = action.integration_id
-    try:
-        service = PagerDutyService.objects.get(id=action.target_identifier)
-    except PagerDutyService.DoesNotExist:
+    organization_id = incident.organization_id
+
+    service: PagerDutyServiceDict | None = None
+    org_integration = integration_service.get_organization_integration(
+        integration_id=integration_id,
+        organization_id=organization_id,
+    )
+    if org_integration is None:
+        org_integration_id = infer_org_integration(integration_id=integration_id, ctx_logger=logger)
```

Comment:
Unrelated to this PR, but, I just read this method. Is it actually safe to infer org integrations like this? Would like to understand that.

Reply:
🤷 @leeandher Has the full story on why this method exists and seems to work.